src/Amazon.Extensions.CognitoAuthentication/CognitoAuthenticationClasses.cs

```diff
@@ -111,19 +111,35 @@ public class InitiateRefreshTokenAuthRequest
     }
 
     /// <summary>
-    /// Class containing the necessary properities to respond to an SMS MFA authentication challenge
+    /// Class containing the necessary properities to respond to an MFA authentication challenge
     /// </summary>
-    public class RespondToSmsMfaRequest
+    public class RespondToMfaRequest
     {
         /// <summary>
         /// The session ID for the current authentication flow.
         /// </summary>
-        public string SessionID { get; set; }
+        public virtual string SessionID { get; set; }
 
         /// <summary>
         /// The MFA verification code needed to authenticate the user.
         /// </summary>
-        public string MfaCode { get; set; }
+        public virtual string MfaCode { get; set; }
+
+        /// <summary>
+        /// The MFA verification code needed to authenticate the user.
+        /// </summary>
+        public virtual ChallengeNameType ChallengeNameType { get; set; }        
+    }
+
+    /// <summary>
+    /// Class containing the necessary properities to respond to an MFA authentication challenge
+    /// </summary>
+    public class RespondToSmsMfaRequest : RespondToMfaRequest
```

Comment:
I suspect we should probably just deprecate this class and encourage users to use the newer `RespondToMfaRequest`. That can be done later though.

Reply:
I also spotted that the override of the property was not set correctly, I will also get that fixed and test the SMS flow again

Reply:
I have now fixed the override of the ChallengeNameType field issue as well on the RespondToSmsMfaRequest

Reply:
I have left the existing class in place for backward compatibility